geth/common/rpccall.go

```diff
@@ -150,12 +159,14 @@ func (r RPCCall) ToSendTxArgs() SendTxArgs {
 		toAddr = gethcommon.HexToAddress("0x0")
 	}
 
-	input := r.ParseData()
+	input := r.ParseInput()
```

Comment:
I believe we could simplify the logic by having `Input` and `Data` always the same value. If both are provided, we can return an error or move forward with `Input` as it is supposed to replace `Data`.

Reply:
Well, theoretically there are a few checks in the `TxQueueManager` for that, so they should never have different values if they are both filled.

> If both are provided, we can return an error or move forward with Input as it is supposed to replace Data.

That exactly what broke DApps in that scenario :-P

Reply:
Hm, wasn't it because we returned only `Input` field? :P I want to always return both. But in case both are provided in the request, we return `Input` as `Data` and `Input` in the response.

Reply:
Hmm, but that is an incoming RPC call, so it can contain anything, right?

Reply:
Right, but I am not sure how this statement affects `Input` or `Data` fields.

Reply:
So, when you send an RPC call with `"data": "my-data"`, you probably expect a response with `"args": { "data": "my-data" }`. If you do it with `input`, you want `input` back.
If we move all the logic here, sure, we will simplify the logic, but we will end up losing the information about what did we receive.
Always fill both `input` and `data` is possible, but I don't know... Can this value be big?

Reply:
Yeah, I'm not sure of the value of filling both `input` and `data`. I prefer the idea of explicitly meeting the requirements of the API, even if the resulting code a bit less concise.

Reply:
That makes sense as well. So, the most important is to stay consistent with go-ethereum. Let's take a closer look how they handle it. I found for instance this: https://sourcegraph.com/github.com/ethereum/go-ethereum@85d5f2c661cd9461e64d6a224076ebc627e14a55/-/blob/internal/ethapi/api.go#L1139-1141

Reply:
but isn't it what we do? we just have a different error message...
https://github.com/status-im/status-go/pull/758/files#diff-a16061196c3bf37b6badd79a450b352aR86

Reply:
I was just making sure. If we do, then great :P